spec/controllers/hq/users_controller_spec.rb

```diff
@@ -31,4 +31,74 @@ def foobar_action
     expect(response).to render_template 'index'
   end
 
+  describe '#edit and #update' do
+    before :each do
+      @old_user = FactoryGirl.build_stubbed(:user)
+      expect(User).to receive(:find).and_return(@old_user)
+    end    
+
+    specify 'editing renders the edit view' do
+      get :edit, id: @old_user.id
+      expect(response).to render_template 'edit'
+    end
+
+    it 'handles invalid updates' do
+      allow(@old_user).to receive(:save).and_return(false)
+      post :update, id: @old_user.id, user: { email: 'some email address'}
+      expect(response).to render_template 'edit'
+      new_user = controller.instance_variable_get(:@user)
```

Comment:
What I was trying to check was that the email parameter sent in the post request actually got set in that entry in the @users variable, so that it appeared when the edit form was re-rendered. How do I use assigns to get at an individual entry in the @users hash? (Again, this was something which wasn't working initially because of the way I was doing the update).